illumio_connector.py

```diff
@@ -374,45 +369,59 @@ def _handle_create_service_binding(self, param):
         workload_hrefs_list = self.handle_comma_seperated_string(
             param["workload_hrefs"]
         )
+        len_workload_list = len(workload_hrefs_list)
         virtual_service_href = param["virtual_service_href"]
 
         ret_val = self.connect_pce(action_result)
         if phantom.is_fail(ret_val):
             return action_result.get_status()
 
         try:
+            service_bindings = self._pce.service_bindings.create(
+                [
+                    {
+                        "workload": {"href": workload_href},
+                        "virtual_service": {"href": virtual_service_href},
+                        "port_overrides": [],
+                    }
+                    for workload_href in workload_hrefs_list
+                ]
+            )
+
             service_binding_list_for_virtual_service = self._pce.service_bindings.get(
                 params={"virtual_service": virtual_service_href}
             )
             bounded_workload_with_virtual_service_list = [
-                service_binding.to_json()["workload"]["href"]
+                service_binding.workload.href
                 for service_binding in service_binding_list_for_virtual_service
             ]
 
             for workload in bounded_workload_with_virtual_service_list:
                 if workload in workload_hrefs_list:
                     workload_hrefs_list.remove(workload)
 
-            if not workload_hrefs_list:
-                return action_result.set_status(
-                    phantom.APP_ERROR, "Workloads already bounded"
+            if workload_hrefs_list:
```

Comment:
No need for the duplicated call to `set_status` in the else block (since both conditions return `APP_SUCCESS` it can be moved after), but maybe this condition isn't needed at all. Is there value to the user in showing them the number of bound workloads or the HREFs of the workloads that failed?

They may want to take action on the failures, so if we're surfacing anything here it should probably be the error messages returned.

Reply:
As the number of bound workloads doesn't matter, we have added a generic "successfully bound workload" message